server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreIT.java

```diff
@@ -277,4 +278,44 @@ public void testRemoteSegmentCleanup() throws Exception {
     public void testRemoteTranslogCleanup() throws Exception {
         verifyRemoteStoreCleanup(true);
     }
+
+    public void testStaleCommitDeletionWithInvokeFlush() throws Exception {
+        internalCluster().startDataOnlyNodes(3);
+        createIndex(INDEX_NAME, remoteStoreIndexSettings(1, 10000l));
+        int numberOfIterations = randomIntBetween(5, 15);
+        indexData(numberOfIterations, true);
+        String indexUUID = client().admin()
+            .indices()
+            .prepareGetSettings(INDEX_NAME)
+            .get()
+            .getSetting(INDEX_NAME, IndexMetadata.SETTING_INDEX_UUID);
+        Path indexPath = Path.of(String.valueOf(absolutePath), indexUUID, "/0/segments/metadata");
+        // Delete is async.
+        assertBusy(() -> {
+            try {
+                int actualFileCount = getFileCount(indexPath);
+                if (numberOfIterations <= RemoteStoreRefreshListener.LAST_N_METADATA_FILES_TO_KEEP) {
+                    assertEquals(numberOfIterations, actualFileCount);
+                } else {
+                    // As delete is async its possible that the file gets created before the deletion or after
+                    // deletion.
+                    assertTrue(actualFileCount >= 10 || actualFileCount <= 11);
+                }
+            } catch (Exception e) {}
```

Comment:
Why do we need to catch the exception here? `getFileCount` method throws exception and if it happens, this test would not fail while it should fail.

Reply:
Yes, this does not seem right. We should change across all instances.